oauth2-authorization-server/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OidcUserInfoTests.java

```diff
@@ -0,0 +1,255 @@
+				.andExpect(jsonPath("middle_name").value("Middle"))
+				.andExpect(jsonPath("nickname").value("User"))
+				.andExpect(jsonPath("preferred_username").value("user"))
+				.andExpect(jsonPath("profile").value("https://example.com/user1"))
+				.andExpect(jsonPath("picture").value("https://example.com/user1.jpg"))
+				.andExpect(jsonPath("website").value("https://example.com"))
+				.andExpect(jsonPath("email").value("user1@example.com"))
+				.andExpect(jsonPath("email_verified").value("true"))
+				.andExpect(jsonPath("gender").value("female"))
+				.andExpect(jsonPath("birthdate").value("1970-01-01"))
+				.andExpect(jsonPath("zoneinfo").value("Europe/Paris"))
+				.andExpect(jsonPath("locale").value("en-US"))
+				.andExpect(jsonPath("phone_number").value("+1 (604) 555-1234;ext=5678"))
+				.andExpect(jsonPath("phone_number_verified").value("false"))
+				.andExpect(jsonPath("address").value("Champ de Mars\n5 Av. Anatole France\n75007 Paris\nFrance"))
+				.andExpect(jsonPath("updated_at").value("1970-01-01T00:00:00Z"));
+		// @formatter:on
+	}
+
+	private OAuth2Authorization createAuthorization() {
+		JoseHeader headers = JoseHeader.withAlgorithm(SignatureAlgorithm.RS256).build();
+		JwtClaimsSet claimSet = JwtClaimsSet.builder().claim(StandardClaimNames.SUB, "user").build();
+		Jwt jwt = this.jwtEncoder.encode(headers, claimSet);
+
+		Instant now = Instant.now();
+		Set<String> scopes = new HashSet<>(OPENID_USER_INFO_SCOPES);
+		OAuth2AccessToken accessToken = new OAuth2AccessToken(
+				OAuth2AccessToken.TokenType.BEARER, jwt.getTokenValue(), now, now.plusSeconds(300), scopes);
+		OidcIdToken idToken = OidcIdToken.withTokenValue("id-token")
+				.claims(claims -> claims.putAll(createUserInfo().getClaims()))
+				.build();
+
+		return TestOAuth2Authorizations.authorization()
+				.accessToken(accessToken)
+				.token(idToken)
+				.build();
+	}
+
+	private static OidcUserInfo createUserInfo() {
+		// @formatter:off
+		return OidcUserInfo.builder()
+				.subject("user1")
+				.name("First Last")
+				.givenName("First")
+				.familyName("Last")
+				.middleName("Middle")
+				.nickname("User")
+				.preferredUsername("user")
+				.profile("https://example.com/user1")
+				.picture("https://example.com/user1.jpg")
+				.website("https://example.com")
+				.email("user1@example.com")
+				.emailVerified(true)
+				.gender("female")
+				.birthdate("1970-01-01")
+				.zoneinfo("Europe/Paris")
+				.locale("en-US")
+				.phoneNumber("+1 (604) 555-1234;ext=5678")
+				.phoneNumberVerified("false")
+				.address("Champ de Mars\n5 Av. Anatole France\n75007 Paris\nFrance")
+				.updatedAt("1970-01-01T00:00:00Z")
+				.build();
+		// @formatter:on
+	}
+
+	@EnableWebSecurity
+	static class AuthorizationServerConfiguration {
+
+		@Bean
+		SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+			OAuth2AuthorizationServerConfigurer<HttpSecurity> authorizationServerConfigurer =
```

Comment:
Let's add another test that configures a custom UserInfo mapper. This test would also serve as an example for users needing to implement custom mapping of the UserInfo response.